File: simple_client.js
// Simple JS client

var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

data_json = {
    location_id: 1,
    wind_speed_knots: 10,
    wind_direction_deg: 11,
    barometric_pressure_hPa: 12.1,
    rain_today_mm: 13,
    rain_rate_mmph: 14.1,
    temperature_cels: 15.1,
    rel_humidity: 0.16,
    uv_index: 17.1,
    heat_index_cels: 18.1
}
const data = JSON.stringify(data_json)

const xhr = new XMLHttpRequest()

xhr.addEventListener('readystatechange', function() {
    if (this.readyState === this.DONE) {
        console.log(this.responseText)
    }
})

xhr.open('POST', 'http://localhost:8080/api/meteo_data', false)
xhr.setRequestHeader('Content-Type', "application/json; charset=utf-8") // Content-Type is CASE SENSITVE!
xhr.send(data)

File: client/simple_client.py
# Simple Python client
import requests
import json

data_json = {
    "location_id": 1,
    "wind_speed_knots": 20,
    "wind_direction_deg": 21,
    "barometric_pressure_ssl_hPa": 22.1,
    "rain_today_mm": 23,
    "rain_rate_mmh": 24.1,
    "temperature_cels": 25.1,
    "rel_humidity": 0.26,
    "uv_index": 27.1,
    "heat_index_cels": 28.1
}

headers = {'Content-Type': 'application/json; charset=utf-8'}
response = requests.post('http://localhost:8080/api/meteo_data', headers = headers, json = data_json)
print("response")
print(response.status_code)
print(response)
